src/services/TransactionService.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace QD\commerce\quickpay\services;
+
+use Craft;
+use craft\base\Component;
+use craft\commerce\models\Transaction as TransactionModel;
+use craft\commerce\Plugin as Commerce;
+use craft\commerce\records\Transaction as TransactionRecord;
+use QD\commerce\quickpay\plugin\Data;
+use craft\helpers\App;
+use Exception;
+
+use craft\commerce\base\Gateway;
+use craft\commerce\db\Table;
+use craft\commerce\elements\Order;
+use craft\commerce\errors\CurrencyException;
+use craft\commerce\errors\OrderStatusException;
+use craft\commerce\errors\TransactionException;
+use craft\commerce\events\TransactionEvent;
+use craft\commerce\helpers\Currency;
+use craft\commerce\models\Transaction;
+use craft\commerce\Plugin;
+use craft\db\Query;
+use craft\errors\ElementNotFoundException;
+use craft\helpers\ArrayHelper;
+use Throwable;
+use yii\base\InvalidConfigException;
+use yii\db\StaleObjectException;
+
+class TransactionService extends Component
+{
+  //* Create
+
+
+  /**
+   * Create a child transaction of the type Authorize
+   *
+   * @param mixed $data
+   * @param TransactionModel $parent
+   * @param string $status
+   * @param string $message
+   * @return TransactionModel
+   */
+  public function createAuthorize(mixed $data, string $status, string $message = ''): TransactionModel
```

Comment:
Mixed should be response model (also other methods in this class) + $parentId as first param